### PR TITLE
Add repo read only

### PR DESCRIPTION
**Pull request by pengpengshui**
Add repo read only module

Modify label issue

Add config> label test case

### CODE REVIEW

File: src/ui_ng/lib/src/tag/tag.component.ts [hunk 1]
```diff
@@ -217,7 +217,13 @@ export class TagComponent implements OnInit, AfterViewInit {
     st.page.size = this.pageSize;
     st.page.from = 0;
     st.page.to = this.pageSize - 1;
-    st.filters = [{property: "name", value: this.lastFilteredTagName}];
+    let selectedLab = this.imageFilterLabels.find(label => label.iconsShow === true);
```
Comment (ninjadq) [inline src/ui_ng/lib/src/tag/tag.component.ts]: The variable name of selectedLab may make some confusion. Consider using full name `selectedLabel` instead?

File: src/ui_ng/lib/src/tag/tag.component.ts [hunk 2]
```diff
@@ -514,6 +520,13 @@ export class TagComponent implements OnInit, AfterViewInit {
                       findedList = BathInfoChanges(findedList, res);
                     });
               }).catch(error => {
+            if (error.status === 503) {
+              Observable.forkJoin(this.translateService.get('BATCH.DELETED_FAILURE'),
+                  this.translateService.get('REPOSITORY.TAGS_NO_DELETE')).subscribe(res => {
+                findedList = BathInfoChanges(findedList, res[0], false, true, res[1]);
```
Comment (ninjadq) [inline src/ui_ng/lib/src/tag/tag.component.ts]: typo in the variable name `BatchInfoChanges`?

Comment (pengpengshui) [reply src/ui_ng/lib/src/tag/tag.component.ts]: The function name was setted BathInfoChanges at the beginning of time,   if changes , may a lot of pages should modiy for the variable name .

File: src/ui_ng/lib/src/tag/tag.component.ts (hunk 1 above)
Comment (pengpengshui) [reply src/ui_ng/lib/src/tag/tag.component.ts]: Thanks for your advice . I will take care for the next time ..

**Pull request merged**